src/containers/PipelineRuns/PipelineRuns.test.js

```diff
@@ -0,0 +1,363 @@
+  const filterInputField = getByTestId('filter-search-bar');
+  fireEvent.change(filterInputField, { target: { value: filterValue } });
+  fireEvent.click(getByText('Add filter'));
+
+  expect(queryByText(filterValue)).toBeTruthy();
+  expect(queryByText('pipelineRunWithSingleLabel')).toBeFalsy();
+  expect(queryByText('pipelineRunWithTwoLabels')).toBeTruthy();
+
+  fireEvent.click(getByText(filterValue));
+  await waitForElement(() => getByText('pipelineRunWithSingleLabel'));
+  expect(queryByText(filterValue)).toBeFalsy();
+
+  expect(queryByText('pipelineRunWithSingleLabel')).toBeTruthy();
+  expect(queryByText('pipelineRunWithTwoLabels')).toBeTruthy();
+});
+
+it('Duplicate label filters are prevented', async () => {
+  const mockTestStore = mockStore(testStore);
+  const match = {
+    params: {},
+    url: '/pipelineruns'
+  };
+
+  const { queryByText, getByTestId, getByText } = renderWithRouter(
+    <Provider store={mockTestStore}>
+      <Route
+        path="/pipelineruns"
+        render={props => (
+          <PipelineRunsContainer
+            {...props}
+            match={match}
+            error={null}
+            loading={false}
+            namespace="namespace-1"
+            fetchPipelineRuns={() => Promise.resolve()}
+            pipelineRuns={pipelineRunsTestStore.pipelineRuns.byId}
+          />
+        )}
+      />
+    </Provider>,
+    { route: '/pipelineruns' }
+  );
+
+  const filterValue = 'baz:bam';
+  const filterInputField = getByTestId('filter-search-bar');
+  fireEvent.change(filterInputField, { target: { value: filterValue } });
+  fireEvent.click(getByText('Add filter'));
+
+  expect(queryByText(filterValue)).toBeTruthy();
+
+  fireEvent.change(filterInputField, { target: { value: filterValue } });
+  fireEvent.click(getByText('Add filter'));
+  expect(queryByText(/No duplicate filters allowed/i)).toBeTruthy();
+});
+
+it('An invalid filter value is disallowed and reported', async () => {
+  const mockTestStore = mockStore(testStore);
+  const match = {
+    params: {},
+    url: '/pipelineruns'
+  };
+
+  const { queryByText, getByTestId, getByText } = renderWithRouter(
+    <Provider store={mockTestStore}>
+      <Route
+        path="/pipelineruns"
+        render={props => (
+          <PipelineRunsContainer
+            {...props}
+            match={match}
+            error={null}
+            loading={false}
+            namespace="namespace-1"
+            fetchPipelineRuns={() => Promise.resolve()}
+            pipelineRuns={pipelineRunsTestStore.pipelineRuns.byId}
+          />
+        )}
+      />
+    </Provider>,
+    { route: '/pipelineruns' }
+  );
+
+  const filterValue = 'baz=bam';
+  const filterInputField = getByTestId('filter-search-bar');
+  fireEvent.change(filterInputField, { target: { value: filterValue } });
+  fireEvent.click(getByText('Add filter'));
+
+  expect(
+    queryByText(
+      /Filters must be of the format labelKey:labelValue and use only accepted label characters/i
```

Comment:
Make sure to include the regex if you go about my suggestion to have as a global var that we can refer to in message

Reply:
going to add a link to the kube label syntax docs instead but will add as global var for easy access 🙂